src/System.Management.Automation/engine/CommandCompletion/CompletionHelpers.cs

```diff
@@ -21,36 +21,66 @@ internal static class CompletionHelpers
         /// <param name="wordToComplete">The word to complete.</param>
         /// <param name="possibleCompletionValues">The possible completion values to iterate.</param>
         /// <param name="toolTipMapping">The optional tool tip mapping delegate.</param>
+        /// <param name="listItemTextMapping">The optional list item text mapping delegate.</param>
         /// <param name="resultType">The optional completion result type. Default is Text.</param>
-        /// <returns></returns>
+        /// <param name="escapeStrategy">
+        /// A delegate that specifies how to escape the word to complete. If null, no escaping is applied.
+        /// </param>
+        /// <returns>List of matching completion results.</returns>
         internal static IEnumerable<CompletionResult> GetMatchingResults(
             string wordToComplete,
             IEnumerable<string> possibleCompletionValues,
             Func<string, string> toolTipMapping = null,
-            CompletionResultType resultType = CompletionResultType.Text)
+            Func<string, string> listItemTextMapping = null,
+            CompletionResultType resultType = CompletionResultType.Text,
+            Func<string, string> escapeStrategy = null)
```

Comment:
We could assign default implementation instead of `null`. 
And more than that finally we have to implement default IsMatch implementation so why not do it now?

Reply:
What would be default implementation if we want to avoid escaping by default?

Reply:
Or do you mean make `IsMatch` a parameter to `GetMatchingResults` as a delegate and then have it use the one which does no escaping by default?

Reply:
Yes, we talked about it. User can use any other method to verify matching, but they don't necessarily need Wildcard.IsMatch().

Reply:
Ok so I think having a `MatchStrategy` delegate is needed. I have added some here: f159a3315826174ad95b71ba289fe947d6922e3d.